### DISCUSSION TITLE
Sending power data immediately after changing the switch (Sonoff POW R3 20D)

### DISCUSSION THREAD

**Discussion by szczukot** [arendst/Tasmota#21273] [Ideas]
When I change switch by mqtt/button/web TELE-SENSOR will be sent unknown when (in TelePeriod window).
But when the button state is changed, all data (voltage, current, power) changes immediately. 
It would be nice if they were also sent immediately.

**Comment by sfromis** [comment]
It cannot be *immediately*, as the energy monitoring needs a bit of time to collect data, but using the `PowerDelta` command, you can have additional data published as energy consumption level steps up/down. This is not directly related to switching the relay on/off.

**Comment by Stolowka** [comment]
Maybe "immediately" is not good word. But energy monitoring collect data less than one second.
That meybe automatic send TELE after 1-2 seconds when switch is toggled, is good option.

Thanks for info about "PowerDelta". This is good workaround for this problem.
But in documenation i see
"PowerDelta will not report when the power turns off. "

**Reply by Stolowka** [reply]
And I see another problem. After set PowerDelta, only tele/sensor is sent, but not domoticz/in.

**Reply by sfromis** [reply]
**If** your hardware supports this, you also have `SetOption21` to still have voltage/frequency available without the relay being on.

**Comment by Stolowka** [comment]
But this still doesn't send mqtt to domoticz/in
Only tele/sensor is works

And another problem: when I set "Update timer" in "domoticz configuration", this is only works only form main idx. Others data (power, currency etc) don't send information so frequently.
